internal/logger/logonce.go

```diff
@@ -41,17 +47,26 @@ func (l *logOnceType) logOnceConsoleIf(ctx context.Context, err error, id string
 	nerr := unwrapErrs(err)
 	l.Lock()
 	shouldLog := true
-	prevErr, ok := l.IDMap[id]
+	prev, ok := l.IDMap[id]
 	if !ok {
-		l.IDMap[id] = nerr
-	} else {
+		l.IDMap[id] = onceErr{
+			Err:   nerr,
+			Count: 1,
+		}
+	} else if prev.Err.Error() == nerr.Error() {
 		// if errors are equal do not log.
-		shouldLog = prevErr.Error() != nerr.Error()
+		prev.Count++
+		l.IDMap[id] = prev
+		shouldLog = false
 	}
 	l.Unlock()
 
 	if shouldLog {
-		consoleLogIf(ctx, err, errKind...)
+		if prev.Count > 1 {
+			consoleLogIf(ctx, fmt.Errorf("Printed %d times, %w", prev.Count, err), errKind...)
```

Comment:
This is a bit strange. It will only print this if `prev.Err.Error() != nerr.Error()`, so it is the same ID, but the error text is different.

So we will get:

```
key "a" does not exist
Printed 2 times, key "b" does not exist
Printed 3 times, key "b" does not exist
Printed 4 times, key "b" does not exist
Printed 5 times, key "b" does not exist
```

Not sure I have a ready-to-go solution, but the above doesn't make sense from a reading perspective.

Reply:
How did you get here? @klauspost the `prev.Count > 1` only if `prev.Err.Error() == nerr.Error()`

Reply:
removed that line, as I need to figure out a better way to bring it back.